lib/util/hud_utility_2024.rb

```diff
@@ -51,14 +51,167 @@ def project_type_number(type)
     project_type_brief(type, true) # reversed
   end
 
+  def residential_project_type_numbers_by_code
+    {
+      ph: [3, 9, 10, 13],
+      rrh: [13],
+      psh: [3],
+      oph: [9, 10],
+      th: [2],
+      es: [0, 1],
+      es_nbn: [1],
+      es_entry_exit: [0],
+      so: [4],
+      sh: [8],
+    }.freeze
+  end
+
+  def performance_reporting
+    { # duplicate of code in various places
+      ph: [3, 9, 10, 13],
+      rrh: [13],
+      psh: [3],
+      oph: [9, 10],
+      th: [2],
+      es: [0, 1],
+      es_nbn: [1],
+      es_entry_exit: [0],
+      so: [4],
+      sh: [8],
+      ce: [14],
+      other: [7],
+      day_shelter: [11],
+      prevention: [12],
+      services_only: [6],
+    }.freeze
+  end
+
+  def project_group_titles
+    {
+      ph: 'Permanent Housing (PH, PSH, & RRH)',
+      es: 'Emergency Shelter (ES)',
```

Comment:
Maybe for consistency it makes sense to list each project type in this group, since it now contains multiple.
```suggestion
      ph: 'Permanent Housing (PH, PSH, & RRH)',
      es: 'Emergency Shelter (ES NBN & ES Entry/Exit)',
```